packages/next/src/client/components/react-dev-overlay/utils/stack-frame.ts

```diff
@@ -8,21 +8,34 @@ import {
   isWebpackInternalResource,
   formatFrameSourceFile,
 } from './webpack-module-path'
-export interface OriginalStackFrame extends OriginalStackFrameResponse {
-  error: boolean
-  reason: string | null
+
+export interface ResolvedOriginalStackFrame extends OriginalStackFrameResponse {
+  error: false
+  reason: null
   external: boolean
   ignored: boolean
   sourceStackFrame: StackFrame
 }
 
+export interface RejectedOriginalStackFrame extends OriginalStackFrameResponse {
+  error: true
+  reason: string
+  external: boolean
+  ignored: boolean
+  sourceStackFrame: StackFrame
+}
+
+export type OriginalStackFrame =
+  | ResolvedOriginalStackFrame
+  | RejectedOriginalStackFrame
+
 function getOriginalStackFrame(
   source: StackFrame,
   response: OriginalStackFrameResponseResult
 ): Promise<OriginalStackFrame> {
-  async function _getOriginalStackFrame(): Promise<OriginalStackFrame> {
+  async function _getOriginalStackFrame(): Promise<ResolvedOriginalStackFrame> {
     if (response.status === 'rejected') {
-      return Promise.reject(new Error(response.reason))
+      throw new Error(response.reason)
```

Comment:
This isn't required. It's equivalent though and just makes it easier to spot that we don't actually return something here that contributes to the return type but rather results in a rejection.